### PR TITLE
Amogh sample label tracking

### PR DESCRIPTION
**Pull request by amogh7joshi**
Adds sample domain tracking

### CODE REVIEW

File: yolo_uda/training/datasets.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+    return img, pad
+
+
+def resize(image, size):
+    image = F.interpolate(image.unsqueeze(0), size=size, mode="nearest").squeeze(0)
+    return image
+
+
+class ImageFolder(Dataset):
```
Comment (mihamerstan) [inline yolo_uda/training/datasets.py:33]: We have 4 different `ImageFolder` classes: this one, `yolov3/utils/datasets.py`, `src/data/image_folder.py` and `src/util/dataset_yolo.py`.  We also have 3 ListDataset classes. Is there a way to use one of those?  If not, can we rename `ListDataset` and `ImageFolder` to help avoid confusion?

File: yolo_uda/training/trainer.py [hunk 2]
```diff
@@ -177,18 +180,20 @@ def train(
         model.train() # set yolo model to training mode
         discriminator.train() # set discriminator to training mode
 
-        for batch_i, (data_source, data_target) in enumerate(
+        for batch_i, contents_ in enumerate(
```
Comment (mihamerstan) [inline yolo_uda/training/trainer.py]: I've never seen the _ at the end of a variable. Was that on purpose?  Fine to leave it if it was, I've just never seen that convention.

File: yolo_uda/training/loader.py [hunk 3]
```diff
@@ -2,47 +2,55 @@
...
+        sample_loc_train += [1] * K
 
     # write to txt file
-    train_output = os.path.join(os.path.dirname(train_path),'train.txt')
-    val_output = os.path.join(os.path.dirname(val_path),'val.txt')
-    paths_output = [val_output, train_output]
+    train_output = os.path.join(os.path.dirname(train_path), 'train.txt')
+    val_output = os.path.join(os.path.dirname(val_path), 'val.txt')
+
+    for fname, loc, paths in zip(
```
Comment (mihamerstan) [inline yolo_uda/training/loader.py]: Smallest edit ever: can you rename this `locs` since you loop through the `locs` later?

**Review by mihamerstan — APPROVED** [review]

File: yolo_uda/training/datasets.py (hunk 1 above)
Comment (amogh7joshi) [reply yolo_uda/training/datasets.py:33]: I think we should rename `ListDataset` then... it's just apparently a pretty common name (along with `ImageFolder`), but I'll rename them to `UDAListDataset` and `UDAImageFolder` just to avoid confusion.

File: yolo_uda/training/trainer.py (hunk 2 above)
Comment (amogh7joshi) [reply yolo_uda/training/trainer.py]: I think it was just a Copilot artifact, I will switch it to `contents` (lol).

Side note: this is a used convention, it's just primarily for cases where the intended variable name is something that's actually used otherwise within python (e.g., iterating over some arbitrary things that you call "objects" would be `for object_ in objects:` instead of `for object in objects:` since `object` is already defined).

File: yolo_uda/training/loader.py (hunk 3 above)
Comment (amogh7joshi) [reply yolo_uda/training/loader.py]: Done

**Comment by amogh7joshi** [comment]
Changes made
